packages/taiko-client/proposer/proposer.go

```diff
@@ -456,24 +451,121 @@ func (p *Proposer) ProposeTxListOntake(
 		return errors.New("insufficient prover balance")
 	}
 
-	txCandidate, err := p.txBuilder.BuildOntake(ctx, txListsBytesArray)
+	txCandidate, cost, err := p.buildCheaperOnTakeTransaction(ctx, txListsBytesArray)
 	if err != nil {
 		log.Warn("Failed to build TaikoL1.proposeBlocksV2 transaction", "error", encoding.TryParsingCustomError(err))
 		return err
 	}
 
+	if p.checkProfitability {
+		profitable, err := p.isProfitable(txLists, cost)
+		if err != nil {
+			return err
+		}
+		if !profitable {
```

Comment:
For this issue https://github.com/taikoxyz/taiko-mono/issues/18085, we want to choose a cheaper way to propose txs rather than stopping the proposal.

Reply:
> For this issue #18085, we want to choose a cheaper way to propose txs rather than stopping the proposal.

Ignore it.